perfmetrics/scripts/run_e2e_tests.sh

```diff
@@ -53,28 +53,162 @@ sudo apt-get install -y gcc python3-dev python3-setuptools
 sudo apt install -y python3-crcmod
 
 # Create bucket for integration tests.
-# The prefix for the random string
-bucketPrefix="gcsfuse-integration-test-"
-# The length of the random string
-length=5
-# Generate the random string
-random_string=$(tr -dc 'a-z0-9' < /dev/urandom | head -c $length)
-BUCKET_NAME=$bucketPrefix$random_string
-echo 'bucket name = '$BUCKET_NAME
-# We are using gcloud alpha because gcloud storage is giving issues running on Kokoro
-gcloud alpha storage buckets create gs://$BUCKET_NAME --project=$PROJECT_ID --location=$BUCKET_LOCATION --uniform-bucket-level-access
+function create_bucket() {
+  bucketPrefix=$1
+  # The length of the random string
+  length=5
+  # Generate the random string
+  random_string=$(tr -dc 'a-z0-9' < /dev/urandom | head -c $length)
+  BUCKET_NAME=$bucketPrefix$random_string
+  # We are using gcloud alpha because gcloud storage is giving issues running on Kokoro
+  gcloud alpha storage buckets create gs://$BUCKET_NAME --project=$PROJECT_ID --location=$BUCKET_LOCATION --uniform-bucket-level-access
+  echo $BUCKET_NAME
+}
+
+# Non parallel execution of integration tests located within specified test directories for group 1.
+function run_non_parallel_tests_group_1() {
+  for test_dir_np_group_1 in "${test_dir_non_parallel_group_1[@]}"
+  do
+    test_path_non_parallel_group_1="./tools/integration_tests/$test_dir_np_group_1"
+    # Executing integration tests
+    GODEBUG=asyncpreemptoff=1 go test $test_path_non_parallel_group_1 -p 1 --integrationTest -v --testbucket=$BUCKET_NAME_NON_PARALLEL_GROUP_1 --testInstalledPackage=$run_e2e_tests_on_package -timeout $INTEGRATION_TEST_TIMEOUT
+    exit_code_non_parallel_group_1=$?
+    if [ $exit_code_non_parallel_group_1 != 0 ]; then
+      test_fail_np_group_1=$exit_code_non_parallel_group_1
```

Comment:
why can't we simply return exit_code_non_parallel_group_1?

Reply:
If we return from inside the loop, the loop will not run for all array packages.